ql/src/java/org/apache/hadoop/hive/ql/ddl/database/drop/DropDatabaseAnalyzer.java

```diff
@@ -73,6 +74,11 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
           }
           outputs.add(new WriteEntity(table, lockType));
         }
+        // fetch all the functions in the database
+        List<Function> functions = db.getFunctionsInDb(databaseName, ".*");
```

Comment:
Given in reality, the number of permanent functions stored in remote db should be smaller compared to the other meta, I think we can use `db.getAllFunctions` then select the functions belonging to this database.

Reply:
On HS2 restart, we load all functions from HMS, 
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java#L364
I think this call is much light weight than the `getAllTableObjects` in this method.
Another point is the `drop database` is a long consuming call as expected and less frequent, I'm a little worry about if it's worth to add an extra API just to fetch all the functions in this database.

Reply:
I still think this method would be useful if we decide to enforce client-side filtering on functions in the future. @nrg4878 -- Can you please share your suggestion here on whether to add a new API for fetching functions in a database or not?